nano/core_test/wallets.cpp

```diff
@@ -87,13 +87,15 @@ TEST (wallets, upgrade)
 		nano::node_init init1;
 		auto node1 (std::make_shared<nano::node> (init1, system.io_ctx, 24001, path, system.alarm, system.logging, system.work));
 		ASSERT_FALSE (init1.error ());
-		node1->wallets.create (id.pub);
```

Comment:
The changes in timing with this PR revealed that by not creating the wallets with `nano::wallets wallets (error, *node1);` it was not creating the environment correctly, therefore it was hitting a failure in mdb_iterator in an unrelated function that acessed the wallets (prioritize frontiers).

Please check that the changes don't invalidate what this test is trying to do.